syntax = "proto3";

package Ydb.Config.V1;

option java_package = "com.yandex.ydb.config.v1";
option java_outer_classname = "ConfigGrpc";
option java_multiple_files = true;

import "ydb/public/api/protos/ydb_config.proto";

service ConfigService {
    // Replace one or many configs.
    rpc ReplaceConfig(Config.ReplaceConfigRequest) returns (Config.ReplaceConfigResponse);

    // Fetch configs from cluster.
    rpc FetchConfig(Config.FetchConfigRequest) returns (Config.FetchConfigResponse);

    // Start cluster initialization process.
    rpc BootstrapCluster(Config.BootstrapClusterRequest) returns (Config.BootstrapClusterResponse);
}
